front-end/src/store/analyzeTranscriptSlice.js

```diff
@@ -31,9 +31,37 @@ export const analyzeTranscriptSlice = createSlice({
     setProjectIdToBeDisplayed: (state, action) => {
       state.projectIdToBeDisplayed = action.payload;
     },
+    toggleStar: (state, action) => {
+      // get the index of the first intent being displayed
+      const firstIndex = state.DisplayingQuestion[state.projectIdToBeDisplayed];
+      // get the transcript that is being displayed and has the indices of the intents
+      const transcript = state.analyzedTranscripts[state.projectIdToBeDisplayed];
+      // change the star state of the intent being changed
+      for (let i = firstIndex; i < firstIndex+3; i++) {
+        if (transcript[i].question === action.payload) {
+          transcript[i].star = !transcript[i].star;
+          return true;
+        }
+        return false;
+      }
+    },
+    toggleFlag: (state, action) => {
+      // get the index of the first intent being displayed
+      const firstIndex = state.DisplayingQuestion[state.projectIdToBeDisplayed];
+      // get the transcript that is being displayed and has the indices of the intents
+      const transcript = state.analyzedTranscripts[state.projectIdToBeDisplayed];
+      // change the bookmark state of the intent being changed
+      for (let i = firstIndex; i < firstIndex+3; i++) {
+        if (transcript[i].question === action.payload) {
+          transcript[i].flag = !transcript[i].flag;
+          return true;
+        }
+        return false;
```

Comment:
Very interesting that you are using return statements! I never thought of using that. We can actually use this in our other reducers too to return If there was an error in changing the redux state. Just a comment, I think you are not using the return statement anywhere so maybe just add the implementation or get rid of it? Or maybe I'm just blind :))

Reply:
This was just to make an early return so that the for-loop stops, but I can find a way of just terminating the loop early!